src/provider/provider_file_memory.c

```diff
@@ -295,6 +295,25 @@ static umf_result_t file_initialize(const void *params, void **provider) {
 
     file_provider->coarse = coarse;
 
+    *provider = file_provider;
+    return UMF_RESULT_SUCCESS;
+
+err_close_fd:
+    utils_close_fd(file_provider->fd);
+err_free_file_provider:
+    umf_ba_global_free(file_provider);
+    return ret;
+}
+
+static umf_result_t file_post_initialize(void *provider) {
+    umf_result_t ret = UMF_RESULT_SUCCESS;
+    file_memory_provider_t *file_provider = provider;
+
+    if (file_provider == NULL) {
```

Comment:
I don't think we have such `if` in other post_init - switch to assert?

Reply:
done